app/forms/declaration/RepresentativeDetails.scala

```diff
@@ -40,8 +40,10 @@ object RepresentativeDetails extends DeclarationPage {
         text().verifying("supplementary.representative.representationType.error.wrongValue", isContainedIn(representativeStatusCodeAllowedValues))
       )
     )(RepresentativeDetails.apply)(RepresentativeDetails.unapply)
-    .verifying("supplementary.namedEntityDetails.error", details => details.details.nonEmpty || details.statusCode.isEmpty)
-    .verifying("supplementary.representative.representationType.error.empty", details => details.details.isEmpty || details.statusCode.nonEmpty)
+    .verifying(
+      "supplementary.representative-details.eori.error",
+      details => (details.statusCode.isEmpty && details.details.map(_.eori).isEmpty) || details.statusCode.nonEmpty
+    )
```

Comment:
This error is related with eori so I think it should be validated no on the model, but on the field. Because with this user after clicking on the error is not able to jump to the input

Reply:
From hints - there could be more then one validation that maps to same error message but have different field to validate on